src/mantle/support/registration/class-post-type-arguments.php

```diff
@@ -0,0 +1,522 @@
+<?php
+/**
+ * Post_Type_Arguments class file
+ *
+ * @package Mantle
+ */
+
+namespace Mantle\Support\Registration;
+
+use Mantle\Contracts\Support\Arrayable;
+use Mantle\Support\Str;
+use Mantle\Support\Traits\Makeable;
+
+use function Mantle\Support\Helpers\collect;
+
+/**
+ * Fluent interface for creating arguments to register a post type.
+ */
+class Post_Type_Arguments implements Arrayable {
+	use Makeable;
+
+	/**
+	 * Arguments to register the post with.
+	 *
+	 * @var array<string, mixed>
+	 */
+	protected array $arguments = [];
+
+	/**
+	 * Get the instance as an array.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function to_array(): array {
+		return $this->arguments;
+	}
+
+	/**
+	 * Get all the arguments.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function all(): array {
+		return $this->arguments;
+	}
+
+	/**
+	 * Set the label for the post type.
+	 *
+	 * @param string      $singular The singular label for the post type.
+	 * @param string|null $plural The plural label for the post type. Optional.
+	 */
+	public function label( string $singular, ?string $plural = null ): static {
+		if ( ! isset( $this->arguments['labels'] ) || ! is_array( $this->arguments['labels'] ) ) {
+			$this->arguments['labels'] = [];
+		}
+
+		if ( ! $plural ) {
+			$plural = Str::plural( $singular );
+		}
+
+		$this->arguments['labels'] = array_merge(
+			$this->arguments['labels'],
+			[
+				'name'                     => $plural,
+				'singular_name'            => $singular,
+				/* translators: %s: Post type name. */
```

Comment:
on the taxonomy, you specified Singular or Plural in the translators comment. That would probably be helpful here too, but not a blocker.